### ISSUE TITLE
"Accept TOS conditions" does not validate.

### ISSUE THREAD

**Issue by mferak** [CakeDC/users#328]
If you're not using Chrome's built-in validation (or if you're using a different browser that doesn't validate fields on a browser level), the Accept TOS conditions field returns always valid. That's because Cake creates a hidden field with a 0 value, thus the current validation conditions always return valid. The validations conditions need to change in order for this to work, or the hiddenField should be removed.

**Comment by mferak** [comment]
This fixes the issue:

<pre>
$validator
            ->add('tos', 'valid', ['rule' => function($data,$provider) {
                if($data==1) {
                    return true;
                }
                return __d('Users', 'Please agree with TOS.');
            }])
            ->requirePresence('tos', 'create')
            ->notEmpty('tos');
</pre>

**Comment by steinkel** [comment]
Yes, this is something we'll need to improve, also to add a way to define the "conditions" page by default...

**Comment by mferak** [comment]
What I did for now was I added this to my validation defaults in my custom model.

**Comment by mferak** [comment]
Unfortunately, this "fix" breaks the automatic creation of a user account by socialLogin. So socialLogin should set "agree" to "1" by default.
